regression/contracts/assigns-enforce-malloc-zero/test.desc

```diff
@@ -0,0 +1,11 @@
+CORE
+main.c
+--enforce-contract foo _ --malloc-may-fail
```

Comment:
You should also include `--malloc-fail-null`.

Reply:
what's the difference  with `--malloc-may-fail` ?

Reply:
I'd expect the same thing, but you need the second flag to ensure that `malloc` will return NULL on failure. See the following example.
```c
// file main.c
#include <stdlib.h>

int main()
{
  int *i = malloc(0);
  assert(__CPROVER_DYNAMIC_OBJECT(i));
  //assert(i == NULL);
}
```

If you run it using `cbmc main.c --malloc-may-fail` the result is:
```shell
** Results:
<builtin-library-malloc> function malloc
[malloc.assertion.1] line 26 max allocation size exceeded: SUCCESS
[malloc.assertion.2] line 31 max allocation may fail: SUCCESS

main.c function main
[main.assertion.1] line 6 assertion IS_DYNAMIC_OBJECT((const void *)i): SUCCESS

** 0 of 3 failed (1 iterations)
VERIFICATION SUCCESSFUL
```

But if you run using `cbmc main.c --malloc-may-fail  --malloc-fail-null --trace`, the result is:
```shell
** Results:
<builtin-library-malloc> function malloc
[malloc.assertion.1] line 26 max allocation size exceeded: SUCCESS
[malloc.assertion.2] line 31 max allocation may fail: SUCCESS

main.c function main
[main.assertion.1] line 6 assertion IS_DYNAMIC_OBJECT((const void *)i): FAILURE

Trace for main.assertion.1:

State 26 file main.c function main line 5 thread 0
----------------------------------------------------
  i=((signed int *)NULL) (00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000)

State 30 file main.c function main line 5 thread 0
----------------------------------------------------
  malloc_size=0ul (00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000)

State 41 file main.c function main line 5 thread 0
----------------------------------------------------
  i=((signed int *)NULL) (00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000)

Violated property:
  file main.c function main line 6 thread 0
  assertion IS_DYNAMIC_OBJECT((const void *)i)
  IS_DYNAMIC_OBJECT((const void *)i)



** 1 of 3 failed (2 iterations)
VERIFICATION FAILED
```

Reply:
The CLI help and doc should be updated to reflect the actual behaviour.